### PR TITLE
AB#426027 Put in toString, shallowCopy, and test for hashCode on dart sealed class enums

### PR DESCRIPTION
**Pull request by dvosscricut**
AB#426027 Put in toString, shallowCopy, and test for hashCode on dart sealed class enums

Dart doesn't support any kind of built in deepCopy so I went with a shallowCopy: https://stackoverflow.com/questions/13107906/how-can-i-clone-an-object-deep-copy-in-dart/67420251#67420251

### CODE REVIEW

File: Sources/FishyJoesCore/Unparse/DartClass.swift [hunk 1]
```diff
@@ -650,7 +650,15 @@ class DartEnumClass: DartClass {
             methods.forEach { output(method: $0, to: fragment) }
 
             fragment.blankLine()
+
             outputNativeMethodDeclarations(to: fragment)
+
+            fragment.blankLine()
+
+            fragment.output("\(unqualifiedName) shallowCopy()", newLineTerminated: false)
+            fragment.outputBlock(" {") {
```
Comment (cobbal) [inline Sources/FishyJoesCore/Unparse/DartClass.swift]: nit, more idiomatic in this codebase to combine the 2 lines:
```suggestion
            fragment.outputBlock("\(unqualifiedName) shallowCopy() {") {
```

File: Sources/FishyJoesCore/Unparse/DartClass.swift [hunk 2]
```diff
@@ -720,6 +728,40 @@ class DartEnumClass: DartClass {
...
+                fragment.outputBlock(" {") {
+                    fragment.output("return \(unqualifiedName).\(enumCase.name)(", newLineTerminated: false)
+                    var paramsString = String()
+                    for (index, value) in enumCase.values.enumerated() {
+                        let valueString = "\(DartClass.deforbidify(value.name))"
+                        paramsString += "\(valueString)"
+                        if index < enumCase.values.indices.upperBound - 1 {
+                            paramsString += ", "
+                        }
+                    }
```
Comment (cobbal) [inline Sources/FishyJoesCore/Unparse/DartClass.swift:747]: nit: `.map(...).joined(", ")` would be a cleaner way to get paramString in both these cases

**Review by cobbal — APPROVED** [review]
Approved with nits

File: integration-tests/TestAPI-bindings/dart/test/enum_test.dart [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import 'dart:js_util';
```
Comment (cobbal) [inline integration-tests/TestAPI-bindings/dart/test/enum_test.dart]: This line is causing CI to fail. It looks related to dart-web? which we don't target

File: Sources/FishyJoesCore/Unparse/DartClass.swift (hunk 2 above)
Comment (dvosscricut) [reply Sources/FishyJoesCore/Unparse/DartClass.swift:747]: outputMap puts it on multiple lines. I wanted it on a single line.

File: integration-tests/TestAPI-bindings/dart/test/enum_test.dart (hunk 3 above)
Comment (dvosscricut) [reply integration-tests/TestAPI-bindings/dart/test/enum_test.dart]: why does it keep adding that? I'll remove it again.

File: Sources/FishyJoesCore/Unparse/DartClass.swift (hunk 2 above)
Comment (cobbal) [reply Sources/FishyJoesCore/Unparse/DartClass.swift:747]: The whole loop could be replaced by `let paramsString = enumCase.values.map { DartClass.deforbidify($0.name) }.joined(separator: ", ")` and the value of paramString would be exactly the same

**Review by mstokercricut — APPROVED** [review]

File: Sources/FishyJoesCore/Unparse/DartClass.swift (hunk 1 above)
Comment (dvosscricut) [reply Sources/FishyJoesCore/Unparse/DartClass.swift]: ok

File: Sources/FishyJoesCore/Unparse/DartClass.swift (hunk 2 above)
Comment (dvosscricut) [reply Sources/FishyJoesCore/Unparse/DartClass.swift:747]: good point

**Pull request merged**